cdc/log.cc

```diff
@@ -1304,6 +1305,15 @@ class transformer final {
                     }
                     return make_lw_shared<cql3::untyped_result_set>(*result_set);
         });
+      } catch (exceptions::unavailable_exception& e) {
+          // `query` can throw `unavailable_exception`, which triggers an error code often interpreted by clients
+          // as ~ "NoHostAvailable". We'll translate it to a `request_execution_exception` and futurize it.
+          cdc_log.debug("Preimage: translating a (read) `unavailable_exception` to `request_execution_exception` - ", e);
+          exceptions::request_execution_exception new_exc(exceptions::exception_code::CDC_PREIMAGE_READ_FAILURE,
+                  format("CDC could not achieve the CL of {}. Required {} responses, got {}.",
+                          e.consistency, e.required, e.alive));
+          return make_exception_future<lw_shared_ptr<cql3::untyped_result_set>>(std::move(new_exc));
```

Comment:
Is the futurization really necessary? Wouldn't just throwing the exception have exactly the same effect?

Reply:
It would be the same (checked experimentally), but I assumed that once futurized it's more "seastary".

Reply:
I think it's better to just throw. It would be simpler and more concise.